### PR TITLE
support iperf-style intermittent results

### PR DESCRIPTION
**Pull request by mxinden**
Introduces an `intermediate` result printed to stdout every 1s.

Required for https://github.com/libp2p/test-plans/issues/261.

### CODE REVIEW

**Comment by mxinden** [comment]
@marten-seemann can you give this pull request a review?

File: client.go [hunk 1]
```diff
@@ -14,7 +14,10 @@ import (
 )
 
 type Result struct {
-	Latency float64 `json:"latency"`
+	Type          string  `json:"type"`
+	TimeSeconds   float64 `json:"timeSeconds"`
+	UploadBytes   uint    `json:"uploadBytes"`
+	DownloadBytes uint    `json:"downloadBytes"`
```
Comment (marten-seemann) [inline client.go:20]: It's better to use uint64 here, otherwise we'll quickly run into problems on 32 bit systems.

File: client.go [hunk 2]
```diff
@@ -64,7 +70,27 @@ func handleClientStream(str io.ReadWriteCloser, uploadBytes, downloadBytes uint6
 	// upload data
 	b = make([]byte, 16*1024)
 	uploadStart := time.Now()
+
+	lastReportTime := time.Now()
+	lastReportWrite := 0
+
 	for uploadBytes > 0 {
+		now := time.Now()
+		if now.Sub(lastReportTime) > time.Second {
```
Comment (marten-seemann) [inline client.go:79]:
```suggestion
		if now.Sub(lastReportTime) >= time.Second {
```

File: client.go [hunk 3]
```diff
@@ -83,12 +111,33 @@ func handleClientStream(str io.ReadWriteCloser, uploadBytes, downloadBytes uint6
 	b = b[:cap(b)]
 	remaining := downloadBytes
 	downloadStart := time.Now()
+
+	lastReportTime = time.Now()
+	lastReportRead := 0
+
 	for remaining > 0 {
+		now := time.Now()
+		if now.Sub(lastReportTime) > time.Second {
```
Comment (marten-seemann) [inline client.go:120]:
```suggestion
		if now.Sub(lastReportTime) >= time.Second {
```

File: client.go [hunk 4]
```diff
@@ -83,12 +111,33 @@ func handleClientStream(str io.ReadWriteCloser, uploadBytes, downloadBytes uint6
 	b = b[:cap(b)]
 	remaining := downloadBytes
 	downloadStart := time.Now()
+
+	lastReportTime = time.Now()
+	lastReportRead := 0
```
Comment (marten-seemann) [inline client.go:116]:
```suggestion
	lastReportRead := uint64(0)
```

File: client.go [hunk 5]
```diff
@@ -64,7 +70,27 @@ func handleClientStream(str io.ReadWriteCloser, uploadBytes, downloadBytes uint6
 	// upload data
 	b = make([]byte, 16*1024)
 	uploadStart := time.Now()
+
+	lastReportTime := time.Now()
+	lastReportWrite := 0
```
Comment (marten-seemann) [inline client.go:75]:
```suggestion
	lastReportWrite := uint64(0)
```

**Comment by mxinden** [comment]
@marten-seemann can you give this another review?

**Review by marten-seemann — APPROVED** [review]

**Comment by mxinden** [comment]
Wow, that was fast. Thank you.

**Pull request merged**